File: src/main/java/com/rkroom/blog/service/PagingService.java
package com.rkroom.blog.service;

import java.util.List;
import java.util.Map;

public interface PagingService {
    List selectAllByPage(int page);
    int selectCountByStatus(boolean status);
    List selectAllArticle(int page);
    List selectArticleByCategoryAndPage(String category,int page);
    int selectPublishedCountByCategory(String category);
    Map selectPreviousArticleSlug(int id, String category);
    Map selectNextArticleSlug(int id,String category);
}

File: src/main/java/com/rkroom/blog/service/PagingService.kt
package com.rkroom.blog.service

interface PagingService {
    fun selectAllByPage(page: Int): List<*>?
    fun selectCountByStatus(status: Boolean): Int
    fun selectAllArticle(page: Int): List<*>?
    fun selectArticleByCategoryAndPage(category: String?, page: Int): List<*>?
    fun selectPublishedCountByCategory(category: String?): Int
    fun selectPreviousArticleSlug(id: Int, category: String?): Map<*, *>?
    fun selectNextArticleSlug(id: Int, category: String?): Map<*, *>?
    fun selectCountAll(): Int
    fun selectQueryByTitle(title:String,page:Int):List<*>?
    fun selectCountQueryByTitle(title: String): Int
}